### PR TITLE
UCAS upload: timestamp format and baseurl/folder env vars

### PR DESCRIPTION
**Pull request by mnacos**
## Context

From recent email correspondence:

> Q: Can you confirm that you can download the files from the testing environment?
> A: File download failed due to "The filename, directory name, or volume label syntax is incorrect.". Can you change filename format like "dfe_apply_itt_applications_20200710_111107_0100.csv" and upload once again via script ?
> 
> Q: Can we set up the production environment as well? What will change? Is it just username/password/folder ID, or also URL?
> A: Yes. Username, Password, URL and FolderID will change.
> 

## Changes proposed in this pull request

Change the format of the timestamp used in csv filenames.

Add new env vars: `UCAS_UPLOAD_BASEURL` and `UCAS_UPLOAD_FOLDER`.

## Guidance to review

Have I added the appropriate env var configuration to all the right templates/pipelines?

## Link to Trello card

[Arrange matching data with UCAS](https://trello.com/c/vX5nXO2g)

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/ucas_matching/upload_matching_data.rb [hunk 1]
```diff
@@ -4,17 +4,14 @@ module UCASMatching
...
-    UPLOAD_FOLDER = 685520099
-
     def perform
       if ENV['UCAS_USERNAME'].blank?
         Rails.logger.info 'UCAS credentials aren\'t configured, assuming that this this is a test environment. Not uploading data.'
         return
       end
 
-      filename = "/tmp/dfe_apply_itt_applications_#{DateTime.now}.csv"
+      timestamp = DateTime.now.strftime('%Y%m%d_%H%M%S_%z').gsub('+', '')
```
Comment (duncanjbrown) [inline app/services/ucas_matching/upload_matching_data.rb:13]: Good thing UCAS doesn't deal with anyone in a negative offset timezone!

File: azure/pipelines/templates/deploy.yml [hunk 2]
```diff
@@ -243,6 +245,8 @@ jobs:
                 -dsiApiSecret "${{parameters.dsiApiSecret}}"
                 -ucasUsername "${{parameters.ucasUsername}}"
                 -ucasPassword "${{parameters.ucasPassword}}"
+                -ucasUploadBaseurl "${{parameters.ucasUploadBaseurl}}"
+                -ucasUploadFolder "${{parameters.ucasUploadFolder}}"
```
Comment (vigneshmsft) [inline azure/pipelines/templates/deploy.yml:249]: no changes are required in `build.yml` and `release.yml` files
and the variable defined in the DevOps group can be directly referenced like
```yaml
-ucasUsername  "$(ucasUsername)"
-ucasPassword "$(ucasPassword)"
-ucasUploadBaseurl "$(ucasUploadBaseurl)"
-ucasUploadFolder "$(ucasUploadFolder)"
```

Comment (duncanjbrown) [reply azure/pipelines/templates/deploy.yml:249]: We need to fix the docs — adding a card: https://trello.com/c/nnCGCFKY/321-update-instructions-for-adding-an-env-variable-in-azure

Comment (mnacos) [reply azure/pipelines/templates/deploy.yml:249]: Taking back changes to `build.yml` and `release.yml'.

Comment (vigneshmsft) [reply azure/pipelines/templates/deploy.yml:249]: cool..in the deploy template file. you need to remove the parameters and just pass the variable using `$()` notation

Comment (mnacos) [reply azure/pipelines/templates/deploy.yml]: You mean like this? https://github.com/DFE-Digital/apply-for-teacher-training/pull/2503/commits/0fef24c1ad4ad11cc08553dc0a72312f85cf325f Should we do this for the other env vars, too?

Comment (vigneshmsft) [reply azure/pipelines/templates/deploy.yml]: yes 👍..other ones need cleaning too, I will take it up.

**Pull request merged**